accounting/ledgerPosting.js

```diff
@@ -52,7 +52,18 @@ module.exports = class LedgerPosting {
         date: this.date || this.reference.date,
         referenceType: referenceType || this.reference.doctype,
         referenceName: referenceName || this.reference.name,
-        description: this.description,
+        description:
+          this.reference.items &&
+          this.reference.items
+            .reduce((acc, curr) => {
+              if (
+                curr.description &&
+                !acc.includes(curr.description.toLowerCase())
+              )
+                acc = acc.concat(curr.description.toLowerCase() + ', ');
+              return acc;
+            }, '')
+            .slice(0, -2),
```

Comment:
@piyushsinghania can you please explain how are we getting the description?

Reply:
yes @ankitsinghaniyaz 
we are using the **reduce** javascript array method to iterate over the items in an invoice. As per the convention for the reduce method, here _**acc is the actuator (previous value)**_, and _**curr is the current value (individual item)**_.

At **_first_**, we are **_passing acc as an empty string_** and **then** with each iteration, we are **_appending the current item description._**